gapic/samplegen/samplegen.py

```diff
@@ -501,8 +503,7 @@ def generate_sample(sample,
                                                          sample["request"])
     v.validate_response(sample["response"])
 
-    sample_id = sample["id"]
-    sample_fpath = sample_id + str(calling_form) + ".py"
+    sample_fpath = sample["id"] + str(calling_form) + ".py"
```

Comment:
IIRC when sample id is unique, that id will be used as the file name. Calling forms are only used to disambiguate when id is not unique (a sample config has more than one valid calling form specified).

@vchudnov-g can confirm if I am wrong.

Reply:
For the time being, since there is no code that iterates over the sample config, this is taking the form of a single boolean flag in the `generate_sample` function. The sample iterating code will do two passes, one to gather counts for ids and the second to generate the samples.